reports/report_builder.py

```diff
@@ -653,24 +659,28 @@ def ws_05(self, ws):
                     counts.update({(r['sex'], TOPIC_GROUPS[r['topic']]): r['n']})
 
             counts_list.append(secondary_counts)
-        self.tabulate_secondary_cols(ws, counts_list[0], self.male_female, MAJOR_TOPICS, row_perc=True)
-        c = ws.dim_colmax + 2
-        self.tabulate_secondary_cols(ws, counts_list[1], self.male_female, MAJOR_TOPICS, row_perc=True, c=c, write_row_headings=False)
-        c = ws.dim_colmax + 2
+        if gen_csv:
+            # Generate CSV
+            generate_csv("presence_of_women_5", ["Topic", "Gender", "Medium", "Count"], counts_list, ws_05_csv)
+        else:
+            self.tabulate_secondary_cols(ws, counts_list[0], self.male_female, MAJOR_TOPICS, row_perc=True)
+            c = ws.dim_colmax + 2
+            self.tabulate_secondary_cols(ws, counts_list[1], self.male_female, MAJOR_TOPICS, row_perc=True, c=c, write_row_headings=False)
+            c = ws.dim_colmax + 2
```

Comment:
I think lets make dataset generation _an optional addition_ rather than a **substitute** i.e. at the end `tabulate` or `tabulate_historical`, check if `with_dataset` is `True`, call `generate_dataset` (or `tabulate_dataset`).

Reply:
Sure, that works @kilemensi But I would say we're already making it optional here ` gen_csv=False`. The reason I would suggest we stick with this is that it prevents us from performing all the calculations inside the `tabulate_` functions.

Reply:
I can see that @esirK but I'm think code clarity rather than optimization (we're only going to generate this once so time shouldn't be a constraint).

If we add it at the end, e.g.
```python
    self.tabulate_secondary_cols(...
    if with_dataset:
         generate_dataset(...
```

Adding dataset generation _does not touch_ the current code at all. Contrast this to what we have now where we need a giant `if/else` change in the middle of every function that needs dataset to separate dataset generation from report generation.

Reply:
Makes sense.